extensions/kafka/src/main/java/io/deephaven/kafka/ingest/GenericRecordChunkAdapter.java

```diff
@@ -120,7 +124,12 @@ private static FieldCopier makeFieldCopier(
                             "Can not map field with non matching logical type to BigDecimal: " +
                                     "field=" + fieldPathStr + ", logical type=" + logicalType);
                 }
-                return new GenericRecordObjectFieldCopier(fieldPathStr, separator, schema);
+                if (dataType.isArray()) {
+                    return new GenericRecordArrayFieldCopier(fieldPathStr, separator, schema, componentType);
+                }
+                if (GenericContainer.class.isAssignableFrom(dataType)) {
```

Comment:
So this is basically `GenericRecord` or anything else?

Reply:
Or GenericArray or GenericEnumSymbol, yeah.